### ISSUE TITLE
Block-based themes without a comments file trigger a depreciation warning

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#28045]
Block-based Themes throw up a depreciation warning if they do not include a `comments.php` file:

> Deprecated: Theme without comments.php is deprecated since version 3.0.0 with no alternative available. Please include a comments.php template in your theme. in /var/www/html/wp-includes/functions.php on line 5059

This happens whether or not the theme uses Gutenberg's comment blocks. To get around it, the theme can supply a `comments.php` file... but if they do, then the file's contents will be used to render the content inside of of the post contents block. So for example, using an empty placeholder file to suppress the error will wipe out the display of comments. 😞

The rendering of comments.php is tracked as part of https://github.com/WordPress/gutenberg/issues/24101, but I'm opening this separate issue to see if we can just suppress that warning for block-based themes. It seems like the warning will continue to be a problem even after #24101 is fixed.

_This came up originally in https://github.com/WordPress/theme-experiments/pull/157._
